core/trino-main/src/main/java/io/trino/operator/project/PageProcessorMetrics.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.project;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.plugin.base.metrics.DurationTiming;
+import io.trino.spi.metrics.Metric;
+import io.trino.spi.metrics.Metrics;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class PageProcessorMetrics
+{
+    private static final String FILTER_TIME = "Filter execution time";
+    private static final String PROJECTION_TIME = "Projection time";
+
+    private long filterTimeNanos;
+    private boolean containsFilter;
+    private double projectionTimeNanos;
+    private boolean containsProjection;
+
+    public void recordFilterTime(long startNanos)
+    {
+        filterTimeNanos += System.nanoTime() - startNanos;
+        containsFilter = true;
+    }
+
+    public void setProjectionTime(double projectionTimeNanos)
```

Comment:
Why `double`?

Reply:
Also, why are the two metrics being tracked differently? (recordXXX vs setXXX). That tells me there's something funny going on with the abstraction.

Reply:
I'm trying re-use the recording of projection execution time by ExpressionProfiler here so that we don't need to call `System.nanoTime` twice for recording start and stop of projection.
I've tweaked the code to avoid using `double` and make the tracking more similar for both fields but it doesn't look exactly the same.

Reply:
Why are we not using the same mechanism for filters? Maybe we should do that first so that both use the same approach.

Reply:
Currently PageProcessor does not perform profiling and tuning of batch sizes based on filter execution time, it does that only for projections. I'm not sure why that is the case or if adding if would be useful. If we need to add that, that seems more appropriate for a separate PR.